sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/TDigestQuantiles.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sketching;
+
+import com.google.auto.value.AutoValue;
+import com.tdunning.math.stats.MergingDigest;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.CustomCoder;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+
+/**
+ * {@code PTransform}s for getting information about quantiles in a stream.
+ *
+ * <p>This class uses the T-Digest structure introduced by Ted Dunning, and more precisely
+ * the {@link MergingDigest} implementation.
+ *
+ * <h2>References</h2>
+ *
+ * <p>The paper and implementation are available on Ted Dunning's
+ * <a href="https://github.com/tdunning/t-digest">Github profile</a>
+ *
+ * <h2>Parameters</h2>
+ *
+ * <p>Only one parameter can be tuned in order ton control the estimation accuracy and memory use,
+ * called the compression factor {@code cf}.
+ *
+ * <p>Stream elements are compressed into a linked list of centroids.
+ * The compression factor is used to limit the number of elements represented by each centroid
+ * as well as the total number of centroids. <br>
+ * The accuracy is really high (in parts per million) for extreme quantiles and
+ * in general {@code <1000 ppm} for middle quantiles. <br>
+ * By default the compression factor is set to 100. A value of 1000 is extremely large.
```

Comment:
Are there any formal guarantees that a given compression factor gives, or any official suggestions on how to set it that you can link to? From this description I'm still not sure what I'll get if I keep it at the default value, and I'm even not sure whether to get greater accuracy I need to make it larger or smaller.

Reply:
I can't find better information in the paper but I can try to explain a little better